### ISSUE TITLE
not working with 3.5 beta

### ISSUE THREAD

**Issue by danielspaniel** [lytics/ember-data-model-fragments#315]
Howdy .. there are errors with using model fragments with ember-data 3.5 beta 3 ( record data enabled ).
  Don't mean to say one line and move on, but it is pretty straightforward ( the issue ) in that I am just making the statement and not pointing out the errors.

**Comment by pjcarly** [comment]
I accidentally upgraded to ED 3.5 (because of a yarn upgrade), I started debugging before I realized that my version was wrong, and this is what I've managed to figure out.

This line is causing an issue.
https://github.com/lytics/ember-data-model-fragments/blob/master/addon/attributes.js#L394
```
function getWithDefault(internalModel, key, options, type) {
  if (key in internalModel._data) { // <--- error starts here
    return internalModel._data[key];
  } else {
    return getDefaultValue(internalModel, options, type);
  }
```

From what I can tell the `internalModel` in ED3.5 doesn't have the `_data` hash anymore, and an exception is thrown saying `Cannot use 'in' operator to search for {key} in undefined`

**Comment by knownasilya** [comment]
@kempt09 and ideas if your branch is close to ready? Maybe open a PR with a WIP to not merge?

**Comment by kempt09** [comment]
@knownasilya yeah I can get to that here this evening. I think I know where the other issue is coming from but I'm not really that deeply knowledgable into the inner workings of ember-data. There is an issue where the fragment not updating after it has been created and then saved. The payload back from the API is not being updated on the fragment attributes on the parent record.

**Comment by sukima** [comment]
This is fixed now?

I mean It is December and Ember-data has moved on to 3.6 Any `ember new` will break with this addon?! Does no one else have a problem with this anymore? I'm confused.

**Comment by sukima** [comment]
Downgrading ember-data to `~3.4.0` fixed the problem for me in case anyone else runs into this problem and tries to Google a fix.

**Comment by luketheobscure** [comment]
Can we get an update on this? Do we need to put out a call for community contributions?

**Comment by jakesjews** [comment]
@luketheobscure I think a call for community contributions would be a good idea. The internal API changed a lot in ember-data 3.5 so I think most of this addon will need to be rewritten. I don't really have the time right now to do it so if someone else is able to that would be amazing.

**Comment by luketheobscure** [comment]
@jakesjews if you're looking for new maintainers, one way to go about that if through the adopted addons program. Take a look at https://github.com/adopted-ember-addons/program-guidelines/blob/master/README.md

**Comment by cohitre** [comment]
I have been trying to get this to work with >=3.5 throughout the last couple of days. Most of the errors seem to be around rollbacks and saving: https://github.com/cohitre/ember-data-model-fragments/pull/1/files

Some of the changes that I've found:
- Using `RecordData` instead of accessing the data directly from the internalModel
- `flushChangedAttributes` was removed
- I keep getting an error about `currentState` when initializing models. Seems like at some point the `writable` value on the property is being dropped

```
integration - Persistence: persisting the owner record when a fragment is dirty moves owner record, fragment array, and all fragments into clean state (3, 4, 7)
fragment is clean
Expected: 	 true
Result: 	false

all fragment array fragments are clean
Expected: 	true
Result: 	false

fragment array is clean
Expected: 	true
Result: 	false
```

**Comment by cohitre** [comment]
I've narrowed down my PR to 7 errors, all related to `currentState`. I think whatever is triggering that is a testing artifact, possibly with how the Store is being setup in the tests.

```
195 tests completed in 10120 milliseconds, with 7 failed, 1 skipped, and 0 todo.
519 assertions of 526 passed, 7 failed.
```

**Comment by cohitre** [comment]
I have gotten all the tests to pass now using EmberData 3.7. I’ll do a couple passes to test with the other versions and fix accordingly.

As it is this PR breaks compatibility with versions before 3.5 and may merit a version bump. 

Any comments or insight into this PR are appreciated.

**Comment by workmanw** [comment]
This issue has been resolved by v4.0.0 of ember-data-model-fragments.
